backend/curation/api/v1/curation/datasets/dataset_id/versions/actions.py

```diff
@@ -13,14 +14,11 @@ def get(dataset_id: str):
     List all versions for a given Dataset that have been published.
     """
     business_logic = get_business_logic()
-    try:
-        UUID(dataset_id)
-    except ValueError as e:
-        raise NotFoundHTTPException("Dataset not found!") from e
+    validate_uuid_else_forbidden(dataset_id)
 
     dataset_versions = business_logic.get_prior_published_versions_for_dataset(DatasetId(dataset_id))
     if len(dataset_versions) == 0:
-        raise NotFoundHTTPException("Dataset not found!")
+        raise ForbiddenHTTPException("Dataset not found!")
```

Comment:
😬 

Perhaps we'll need to talk. Maybe I am forgetting something.

Reply:
I was trying to be consistent--wherever we call get_inferred_{collection/dataset}_else_forbidden, we return 403 Forbidden when the ID does not find an entity

Reply:
correction: just double checked, we only have get_inferred_collection_version_else_forbidden. Still, seems strange to return Forbidden for Collections Versions and Not Founds for Dataset Versions, albeit less inconsistent than I initially thought since they're different entities.